### PR TITLE
collect additional metadata for buckets and objects in aws driver

### PR DESCRIPTION
**Pull request by devanshjain7**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
Added additional metadata collection for buckets and objects in aws driver
The following metadata fields have been added:

**Bucket Level:**
Bucket Access Control List

**Object Level:**
Object Access Control List
Redirect Location
Object Tags
Metadata

**Test Report Added?**:
/kind TESTED

**Test Report**:
image

image

### CODE REVIEW

File: metadata/pkg/drivers/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package utils
+
+import (
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/opensds/multi-cloud/metadata/pkg/model"
+)
+
+func AclMapper(grant *s3.Grant) *model.Access {
```
Comment (josephjacobmorris) [inline metadata/pkg/drivers/utils/utils.go:8]: can maybe move to the utils directory in the metadata folder level

**Review by josephjacobmorris — CHANGES_REQUESTED** [review]

File: metadata/pkg/drivers/aws/aws.go [hunk 2]
```diff
@@ -69,28 +70,58 @@ func ObjectList(sess *session.Session, bucket *model.MetaBucket) error {
 		totSize += obj.Size
 		obj.StorageClass = *object.StorageClass
 
+		tags, err := svc.GetObjectTagging(&s3.GetObjectTaggingInput{Bucket: &bucket.Name, Key: &obj.ObjectName})
```
Comment (PravinRanjan10) [inline metadata/pkg/drivers/aws/aws.go:73]: please move to separate function, as original function body getting too long.

File: metadata/pkg/drivers/aws/aws.go [hunk 3]
```diff
@@ -69,28 +70,58 @@ func ObjectList(sess *session.Session, bucket *model.MetaBucket) error {
...
+
+		if err == nil {
+			tagset := make(map[string]string)
```
Comment (PravinRanjan10) [inline metadata/pkg/drivers/aws/aws.go:76]: make may not required.

File: metadata/pkg/drivers/aws/aws.go [hunk 4]
```diff
@@ -134,7 +167,17 @@ func GetBucketMeta(buckIdx int, bucket *s3.Bucket, sess *session.Session, bucket
 	} else if !strings.Contains(err.Error(), "NoSuchTagSet") {
 		log.Errorf("unable to get bucket tags. failed with error: %v", err)
 	}
-	bucketArray[buckIdx] = buck
+
+	acl, err := svc.GetBucketAcl(&s3.GetBucketAclInput{Bucket: bucket.Name})
+	if err != nil {
+		log.Errorf("unable to get bucket Acl. failed with error: %v", err)
+	} else {
+		access := make([]*model.Access, len(acl.Grants))
```
Comment (PravinRanjan10) [inline metadata/pkg/drivers/aws/aws.go:175]: make may not required

File: metadata/pkg/drivers/utils/utils.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
+package utils
```
Comment (PravinRanjan10) [inline metadata/pkg/drivers/utils/utils.go:1]: please add headers. You may refer from some other files

**Review by PravinRanjan10 — CHANGES_REQUESTED** [review]

File: metadata/pkg/drivers/utils/utils.go (hunk 1 above)
Comment (devanshjain7) [reply metadata/pkg/drivers/utils/utils.go]: done

File: metadata/pkg/drivers/utils/utils.go (hunk 5 above)
Comment (devanshjain7) [reply metadata/pkg/drivers/utils/utils.go]: moved to metadata/pkg/utils

File: metadata/pkg/drivers/aws/aws.go (hunk 4 above)
Comment (devanshjain7) [reply metadata/pkg/drivers/aws/aws.go:175]: make is being used for map initialisation.

File: metadata/pkg/drivers/aws/aws.go (hunk 3 above)
Comment (devanshjain7) [reply metadata/pkg/drivers/aws/aws.go:76]: make is being used for map initialisation.

File: metadata/pkg/drivers/aws/aws.go (hunk 2 above)
Comment (devanshjain7) [reply metadata/pkg/drivers/aws/aws.go:105]: done

**Review by PravinRanjan10 — APPROVED** [review]
LGTM

**Pull request merged**